Source/WebKit/Platform/IPC/Connection.cpp

```diff
@@ -1335,14 +1335,18 @@ void Connection::enqueueIncomingMessage(UniqueRef<Decoder> incomingMessage)
             return;
 
         if (isIncomingMessagesThrottlingEnabled() && m_incomingMessages.size() >= maxPendingIncomingMessagesKillingThreshold) {
-            dispatchToClientWithIncomingMessagesLock([protectedThis = Ref { *this }] {
-                if (!protectedThis->m_client)
-                    return;
-                protectedThis->m_client->requestRemoteProcessTermination();
-                RELEASE_LOG_FAULT(IPC, "%p - Connection::enqueueIncomingMessage: Over %zu incoming messages have been queued without the main thread processing them, terminating the remote process as it seems to be misbehaving", protectedThis.ptr(), maxPendingIncomingMessagesKillingThreshold);
-                Locker lock { protectedThis->m_incomingMessagesLock };
-                protectedThis->m_incomingMessages.clear();
-            });
+            if (m_incomingMessages.size() == maxPendingIncomingMessagesKillingThreshold) {
+                dispatchToClientWithIncomingMessagesLock([protectedThis = Ref { *this }] {
+                    if (!protectedThis->m_client)
+                        return;
+                    protectedThis->m_client->requestRemoteProcessTermination();
+                    if (!protectedThis->m_wasKilled)
+                        protectedThis->kill();
```

Comment:
Is this function available when `USE(EXTENSIONKIT_PROCESS_TERMINATION)`?

Reply:
Yes, good point. I'll re-work the patch.